runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -268,6 +276,16 @@ public V get(@ConflictParameter Object key) {
             entryStream = mainMap.entrySet().parallelStream();
             log.warn("getByIndexAndFilter: Attempted getByIndexAndFilter without indexing");
         } else {
+            // Collect the indexes first
+            try {
+                // Submitting this empty job and waiting for it will cause us
+                // to wait for all pending jobs. Since we're guaranteed to be
+                // single threaded otherwise, this should be safe.
+                indexExecutor.submit(() -> null).get();
+            } catch (Exception e) {
+                // Rethrow exception
+                throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)